subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/fixtures/JacocoCoverage.groovy

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.jacoco.plugins.fixtures
+
+final class JacocoCoverage {
+
+    JacocoCoverage() {}
+
+    final static String[] ALL = ['0.6.0.201210061924', '0.6.2.201302030002', '0.6.3.201306030806', '0.7.1.201405082137', '0.7.6.201602180812'].asImmutable()
+
+    final static List<String> COVERAGE_CHECK_SUPPORTED = ALL.findAll {
+        def jacocoVersion = new JacocoVersion(it)
+        def supportedJacocoVersion = JacocoVersion.CHECK_INTRODUCED
+        jacocoVersion.compareTo(supportedJacocoVersion) >= 0
+    }.asImmutable()
+
+    final static List<String> COVERAGE_CHECK_UNSUPPORTED = ALL.findAll {
+        def jacocoVersion = new JacocoVersion(it)
+        def supportedJacocoVersion = JacocoVersion.CHECK_INTRODUCED
+        jacocoVersion.compareTo(supportedJacocoVersion) == -1
+    }.asImmutable()
+
+    private static class JacocoVersion implements Comparable<JacocoVersion> {
+        final static CHECK_INTRODUCED = new JacocoVersion(0, 6, 3)
```

Comment:
That's a "pattern" described in [Effective Java](https://www.amazon.com/Effective-Java-2nd-Joshua-Bloch/dp/0321356683) on page 76, chapter 4. What would be benefit of moving it out?